pkg/insights/insightsreport/insightsreport.go

```diff
@@ -152,11 +165,14 @@ func (r *Gatherer) RetrieveReport() {
 			}
 
 			firstPullDone = true
-			if !delayTimer.Stop() {
-				<-delayTimer.C
```

Comment:
It's still not clear to me whether we need to call this @joselsegura. This part of code was almost never (with waiting for clusterversion exception) called before AFAIK. 
My thinking is following. We already drained the channel in this moment and thus the `delayTimer.Stop()` returns `false` (see https://golang.org/pkg/time/#Timer.Stop), which leads to blocking call of `<-delayTimer.C`. Maybe it should be:
```
if delayTimer.Stop() {
	<-delayTimer.C
}
```
or maybe I am completely wrong....

Reply:
TBH I wrote it a long time ago and I'm not 100% sure, but according to the example in the link to Golang `time` library, the `Stop` call will return "false" «if the timer has already expired or been stopped. Stop does not close the channel, to prevent a read from the channel succeeding incorrectly.»

So I think this logic is correct: if `Stop` returns "true", you are done with it. If it returns "false", you need to read from the channel to ensure anybody else is consuming it (if I am understanding the library reference correctly!).

It is probably a very rare corner case, so probably your solution will work much better

Reply:
The documentation for `Reset` describes it even better IMO https://golang.org/pkg/time/#Timer.Reset and I think we don't need to call/check `Stop`, because it will always return `false` in this situation, because the timer has already expired (as per doc).